src/main/java/net/minecraftforge/items/pickup/ItemPickupReason.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.items.pickup;
+
+import com.google.common.collect.Maps;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+import net.minecraft.util.StringRepresentable;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class representing a reason in which an ItemEntity is being picked up.
+ * @see ItemPickupReasons
+ */
+public final class ItemPickupReason implements StringRepresentable
+{
```

Comment:
Couldn't this entire class be replaced with `record ItemPickupReason(String name){}`
Records implement the equals/hashcodes/ for us.
Tho we would loose the instance equality because there is no way to make the constructor private. 
However we are using them in sets, which use hashCode/equals anyway, so there is no performance difference.
Also, why are you implementing StringRepresentable?

Reply:
> Also, why are you implementing StringRepresentable?
tbh I have gotten into a bit of a bad habit with this interface, if an object has a String property representing its internal/serialized name, I tend to implement that interface without thinking, usually though it's for enums/block states which makes sense there.